API/hermes/JNI_OnLoad.cpp

```diff
@@ -0,0 +1,5 @@
+#include <fbjni/fbjni.h>
+
+extern "C" jint JNIEXPORT JNICALL JNI_OnLoad(JavaVM *jvm, void *reserved) {
```

Comment:
This is here presumably to implement "Ensure fbjni is initialized when loading libhermes.so."  But this is a bit surprising to me, as hermes.so isn't directly loaded by Java, so I would not think this would be called.

I'm guessing you did this to fix a problem of some sort, but it's not clear what the problem was.  Can you say what you were trying to fix with this?

Reply:
Hi @mhorowitz 

As we discussed offline, it was required because hermes is not built to depend on libfbjni.so, but instead directly compiling the fbjni c++ sources into the libhermes.so. This results in libhermes.so having a private copy of the jvm pointer.

But, later when I tried with react-native, the fbjni calls to Java code seems to work even without this initialization. Hence, there is something going on which i haven't understood well enough.

Please let me know if you want me to try to look into the details here.

Reply:
@mganandraj I think your observation about libhermes including its own fbjni is on point and relevant.  I think the correct long-term fix is for Hermes to depend on prebuilt external fbjni from maven as RN does, so they can both share the same libfbjni.so.  But this is not a simple fix, so for now, including JNI_OnLoad in libhermes is an ok workaround, but we should try to fix this for real, too.